VOCABULARY-CREATION-GUIDE.md

```diff
@@ -2,4 +2,33 @@
 
 Thing Description Forms can be extended with annotations on how to activate affordances.
 These annotations are typically from protocol vocabularies that need to exist separately from the binding template document.
-This document gives guidance how such vocabularies can be created and how they should be used together with the templates and tools provided.
\ No newline at end of file
+This document gives guidance how such vocabularies can be created and how they should be used together with the templates and tools provided.
+
+## Components of a Binding Template
+
+Besides the actual binding template document that is to be used by implementors
+(the `index.html` file), there are several additional companion documents which
+enable the binding to be used in practice.
+To enable TD consumers to work with the added vocabulary, the definition of a
+JSON-LD context file (`context.jsonld`) is necessary.
+This file maps the TD-specific vocabulary defined in the binding template to an
+OWL ontology (`ontology.ttl`), enabling the use of Semantic Web technologies.
+Finally, a JSON schema file (`example-protocol.schema.json`) should be provided
+to enable automatic validation of forms using the binding template.
+
+Building upon this set of documents, an automated rendering process of the
+binding template document from the ontology can be established.
+To do so, a `template.sparql` file needs to be created which can insert rendered
+tables into placeholders (`%s`) within an `index.template.index` file.
```

Comment:
Would this work for you, @mahdanoura? Feel free to make another suggestion otherwise.

```suggestion
To do so, a `template.sparql` file needs to be created based on
[STTL](https://ns.inria.fr/sparql-template/), which defines rules for transforming a source RDF graph into a result text.
The rules are applied to an `index.template.html` file which contains placeholders
(`%s`) where the resulting tables are inserted.
```